t/plugin/proxy-cache/memory.t

```diff
@@ -661,3 +661,57 @@ GET /hello
 --- more_headers
 Cache-Control: only-if-cached
 --- error_code: 504
+
+
+
+=== TEST 36: configure plugin without memory_cache zone
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                        "plugins": {
+                            "proxy-cache": {
+                               "cache_strategy": "memory",
+                               "cache_key":["$host","$uri"],
+                               "cache_bypass": ["$arg_bypass"],
+                               "cache_control": true,
+                               "cache_method": ["GET"],
+                               "cache_ttl": 10,
+                               "cache_http_status": [200]
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1986": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 200
+--- response_body
+passed
+
+
+
+=== TEST 37: hit route
+--- request
+GET /hello
+--- grep_error_log eval
+qr/no cache_zone provided/
+--- grep_error_log_out
+no cache_zone provided
+no cache_zone provided
```

Comment:
@kayx23 Does this meet your requirement?

Reply:
There might be a misunderstanding? This should actually error out with an error in the 400 range and with the response body reminding users they should configure `cache_zone` to be an in-memory cache zone.

Current behaviour: If you configure `proxy-cache` plugin with `"cache_strategy": "memory"` and without `cache_zone`, currently you get a 500 Internal Server Error.

Reply:
@kayx23 That internal server error takes place when the requests are received and not when plugin is configured in admin API.
Now instead of 500 internal server error, the error logs will point out the fact that cache_zone is not configured so proxy-cache won't work as expected. The requests still pass through. 

I think your suggestion is more in line with, rejecting the configuration during schema validation if cache_zone is explicitly not provided while cache_strategy is `memory`